network-protocols.md

```diff
@@ -62,6 +62,34 @@ func SayHello(p PeerID) {
 
 Upon receiving a 'hello' stream from another node, you should read off the CBOR RPC message, and then check that the genesis hash matches you genesis hash. If it does not, that node is not part of your network, and should probably be disconnected from. Next, the `HeaviestTipSet`, claimed `HeaviestTipSetHeight`, and peerID of the other node should be passed to the chain sync subsystem.
 
+# Heartbeat
+
+The Heartbeat protocol is used to ensure that nodes in a Filecoin node's PeerSet is still running. The libp2p protocol ID for this protocol is `/fil/heartbeat/1.0.0`.
```

Comment:
What is this for exactly? libp2p already gives us whether or not peers are still online and connected, we get notifications from libp2p when peers connect and disconnect.

Reply:
Perfect. I didn't know this. Is there some sort of callback or event-listener we can put on this in libp2p: would be useful to ensure that nodes get cycles out of the peerset appropriately as they become unresponsive (in addition to doing it naturally to ensure you have wide network connectivity)

Reply:
ie I want a replacement node every time one times out.

Reply:
Yeah, you can register for a disconnect notification.

Reply:
The callback for disconnect notifications gives you which peer disconnected, so you just say "tell me about all the disconnects that happen" and its up to you to determine what to do about it.